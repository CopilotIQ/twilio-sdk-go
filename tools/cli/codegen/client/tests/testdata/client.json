{
  "packageName": "v1",
  "name": "Proxy",
  "config": {
    "beta": true,
    "subDomain": "proxy",
    "apiVersion": "v1"
  },
  "imports": [
    "github.com/RJPearson94/twilio-sdk-go/service/proxy/v1/service",
    "github.com/RJPearson94/twilio-sdk-go/service/proxy/v1/services"
  ],
  "subClients": [
    {
      "name": "Services"
    },
    {
      "name": "Service",
      "functionParams": [
        {
          "name": "serviceSid",
          "type": "string"
        }
      ]
    }
  ]
}
