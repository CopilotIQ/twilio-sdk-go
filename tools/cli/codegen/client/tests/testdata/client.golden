// This is an autogenerated file. DO NOT MODIFY
package v1

import (
	"github.com/RJPearson94/twilio-sdk-go/client"
	"github.com/RJPearson94/twilio-sdk-go/service/proxy/v1/service"
	"github.com/RJPearson94/twilio-sdk-go/service/proxy/v1/services"
	"github.com/RJPearson94/twilio-sdk-go/session"
	"github.com/RJPearson94/twilio-sdk-go/session/credentials"
)

type Proxy struct {
	client *client.Client

	Services *services.Client
	Service  func(string) *service.Client
}

// Used for testing purposes only
func (s Proxy) GetClient() *client.Client {
	return s.client
}

func New(sess *session.Session) *Proxy {
	config := client.GetDefaultConfig()
	config.Beta = true
	config.SubDomain = "proxy"
	config.APIVersion = "v1"

	return NewWithClient(client.New(sess, config))
}

func NewWithClient(client *client.Client) *Proxy {
	return &Proxy{
		client: client,

		Services: services.New(client),
		Service:  func(serviceSid string) *service.Client { return service.New(client, serviceSid) },
	}
}

func NewWithCredentials(creds *credentials.Credentials) *Proxy {
	return New(session.New(creds))
}
